File: dotnet/Model/Span.cs
namespace Philter.Model
{
    public class Span
    { 

        private int _characterStart;
        private int _characterEnd;
        private string _context;
        private string _documentId;
        private double _confidence;
        private string _replacement;
        private FilterType _filterType;

        public int characterStart
        {
            get
            {
                return this._characterStart;
            }
            set
            {
                this._characterStart = value;
            }
        }

        public int characterEnd
        {
            get
            {
                return this._characterEnd;
            }
            set
            {
                this._characterEnd = value;
            }
        }

        public string context
        {
            get
            {
                return this._context;
            }
            set
            {
                this._context = value;
            }
        }

        public string documentId
        {
            get
            {
                return this._documentId;
            }
            set
            {
                this._documentId = value;
            }
        }

        public double confidence
        {
            get
            {
                return this._confidence;
            }
            set
            {
                this._confidence = value;
            }
        }

        public string replacement
        {
            get
            {
                return this._replacement;
            }
            set
            {
                this._replacement = value;
            }
        }

        public FilterType filterType
        {
            get
            {
                return this._filterType;
            }
            set
            {
                this._filterType = value;
            }
        }

    }
}

File: src/main/java/com/mtnfog/philter/sdk/model/Span.java
/*******************************************************************************
 * Copyright 2020 Mountain Fog, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License.  You may obtain a copy
 * of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations under
 * the License.
 ******************************************************************************/
package com.mtnfog.philter.sdk.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Identifies a section of text that Philter found to contain sensitive information.
 */
public class Span {

    @Expose
    @SerializedName("id")
    private String id;

    @Expose
    @SerializedName("characterStart")
    private int characterStart;

    @Expose
    @SerializedName("characterEnd")
    private int characterEnd;

    @Expose
    @SerializedName("filterType")
    private String filterType;

    @Expose
    @SerializedName("context")
    private String context;

    @Expose
    @SerializedName("documentId")
    private String documentId;

    @Expose
    @SerializedName("confidence")
    private double confidence;

    @Expose
    @SerializedName("text")
    private String text;

    @Expose
    @SerializedName("replacement")
    private String replacement;

    @Expose
    @SerializedName("ignored")
    private boolean ignored;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public int getCharacterStart() {
        return characterStart;
    }

    public void setCharacterStart(int characterStart) {
        this.characterStart = characterStart;
    }

    public int getCharacterEnd() {
        return characterEnd;
    }

    public void setCharacterEnd(int characterEnd) {
        this.characterEnd = characterEnd;
    }

    public String getFilterType() {
        return filterType;
    }

    public void setFilterType(String filterType) {
        this.filterType = filterType;
    }

    public String getContext() {
        return context;
    }

    public void setContext(String context) {
        this.context = context;
    }

    public String getDocumentId() {
        return documentId;
    }

    public void setDocumentId(String documentId) {
        this.documentId = documentId;
    }

    public double getConfidence() {
        return confidence;
    }

    public void setConfidence(double confidence) {
        this.confidence = confidence;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getReplacement() {
        return replacement;
    }

    public void setReplacement(String replacement) {
        this.replacement = replacement;
    }

    public boolean isIgnored() {
        return ignored;
    }

    public void setIgnored(boolean ignored) {
        this.ignored = ignored;
    }

}
